core/lib/zksync_core/src/consensus/proto/mod.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package zksync.zksync_core.consensus;
+
+message Transaction {
+  // Default derive(serde::Serialize) encoding of the zksync_types::Transaction.
+  // TODO(gprusak): it is neither efficient, nor suitable for version control.
+  //     replace with a more robust encoding.
+  optional string json = 1; // required
+}
+
+message Payload {
+  optional bytes hash = 1; // required; H256
+  optional uint32 l1_batch_number = 2; // required
+  optional uint64 timestamp = 3; // required; ms since UNIX epoch?
```

Comment:
As defined now, `SyncBlock.timestamp` is the number of *seconds* since epoch. This imprecise timestamp works because miniblock timestamps must be distinct as per EVM restrictions.

Reply:
thx, where it is defined though?

Reply:
It is equivalent to the `timestamp` field in the `miniblocks` Postgres table and this to [`MiniblockHeader.timestamp`](https://github.com/matter-labs/zksync-era/blob/f4313a4e5a67f616d2dfa8f364c47cb73cef1ec7/core/lib/types/src/block.rs#L74). It maps to multiple fields in the state keeper / VM types (e.g., [`L2BlockEnv.timestamp`](https://github.com/matter-labs/zksync-era/blob/f4313a4e5a67f616d2dfa8f364c47cb73cef1ec7/core/lib/multivm/src/interface/types/inputs/l2_block.rs#L6) and [`MiniblockParams.timestamp`](https://github.com/matter-labs/zksync-era/blob/f4313a4e5a67f616d2dfa8f364c47cb73cef1ec7/core/lib/zksync_core/src/state_keeper/io/mod.rs#L53)). AFAICT, the fact that it's seconds since epoch is implicit / tribal knowledge; see e.g. [`sleep_past`](https://github.com/matter-labs/zksync-era/blob/f4313a4e5a67f616d2dfa8f364c47cb73cef1ec7/core/lib/zksync_core/src/state_keeper/io/mempool.rs#L357) implementation. Also, there's a fact that EVM block timestamps are measured in seconds. I think that it can be quite confusing, and optimally timestamps should have a newtype wrapper.